petition_service/internal/controller/rpc/petition_controller.go

```diff
@@ -96,10 +97,21 @@ func (s *Server) CreateVote(_ context.Context, req *pb.CreateVoteRequest) (*empt
 		PetitionID: uint(req.PetitionId),
 		UserID:     uint(req.UserId),
 	}
+
 	err := s.PetitionService.CreateVote(newVote)
 	if err != nil {
 		return nil, err
 	}
+	petition, err := s.PetitionService.GetByID(uint(req.PetitionId))
+	if err != nil {
+		return nil, err
+	}
+	currVotes := petition.CurrVotes
+	currVotes++
+	err = s.PetitionService.UpdateCurrVotes(uint(req.PetitionId), currVotes)
```

Comment:
You get a petition by id in the controller, only to pass its id to UpdateCurrVotes, where you retrieve the petition from the db again. Maybe don't increment currVotes in the controller, and instead have a function in the service that increments the curr votes by one?